### ISSUE TITLE
Timeout on test

### ISSUE THREAD

**Issue by gerhardsletten** [danielstjules/mocha.parallel#5]
If you need to increase the timeout of a test, in a regular suite you are able to [set a timeout](https://mochajs.org/#timeouts), but this faild with parallel:

```
parallel('Test with long timeout', function() {
	it('Test running in 3000ms', function (done) {
		this.timeout(4000);
		setTimeout(function(){
			done();
		},3000)
	});
});
```

**Comment by danielstjules** [comment]
Yea, sorry. Dynamically skipping specs also aren't supported. E.g.
```
it('test', function() { this.skip(); });
```

Only the following is currently handled:
```
 *                                               ... Supports hooks, pending
 * or skipped specs/suites via parallel.skip() and it.skip(), but not nested
 * suites.  parallel.only() and it.only() may be used to only wait on the
 * specified specs and suites.
 *
```

**Comment by danielstjules** [comment]
Added basic support for contexts in https://github.com/danielstjules/mocha.parallel/commit/a5ad65119f2607568db8ce3c309905ad5e6156b5 It works for this.skip(), but only partially for this.timeout()

this.timeout() https://github.com/danielstjules/mocha.parallel/blob/a5ad65119f2607568db8ce3c309905ad5e6156b5/fixtures/contextTimeout.js works correctly in mocha 2.2.5, but not mocha 2.3.x. This is because the timeout error isn't thrown, but instead passed directly to the done callback. To avoid that, I'd need to replace runnable.resetTimeout: https://github.com/mochajs/mocha/blob/master/lib/runnable.js#L177-L192 Unfortunately I need access to the runnable via `ctx._runnable`, but it doesn't exist during spec definition - only execution. https://github.com/mochajs/mocha/blob/master/lib/runnable.js#L219
``` javascript
  // Sometimes the ctx exists, but it is not runnable
  if (ctx && ctx.runnable) {
    ctx.runnable(this);
  }
```

So it doesn't seem possible to correctly support with mocha 2.3.x. If used with those versions, the exception will be associated with the beforeEach hook rather than the spec. But it works with 2.2.5 (only other version I tested).

This entire library is a hack, so it was pretty likely it'd hit barriers like this one :)
